### PR TITLE
Further Object testing

### PR DESCRIPTION
**Pull request by geenen124**
Extracted XML constructors to the respective objects

### CODE REVIEW

File: test/objects/Camera.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+                    name: [ "70D" ],
+                    description: [ "Canon" ],
+                    movementMargin: [ 2 ],
+                },
+            ]
+        };
+
+        const camera = Camera.fromXML(xml);
+        expect(camera).to.not.be.null;
+        expect(camera.name).to.equal("Left of Center");
```
Comment (bart-de-jonge) [inline test/objects/Camera.js:29]: You only test for the name here, shouldn''t you test for the description, movementMargin and cameraType as well?

File: test/objects/CameraType.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    it("Should Create CameraType Object from valid XML", done => {
+        const xml = {
+            name: ["70D"],
+            description: ["Canon"],
+            movementMargin: [2],
+        };
+
+        const cameraType = CameraType.fromXML(xml);
+        expect(cameraType).to.not.be.null;
+        expect(cameraType.name).to.equal("70D");
```
Comment (bart-de-jonge) [inline test/objects/CameraType.js:20]: Same here, test for description and movementMargin

**Comment by bart-de-jonge** [comment]
2 small comments, other than that great work (including the extraction to builder methods!)

**Comment by mennoo1996** [comment]
Agree with bart's comments, rest looks fine

**Comment by geenen124** [comment]
Also added some socket.io message passing

**Comment by mennoo1996** [comment]
Nice work!

**Pull request merged**